### PR TITLE
partial variables

### PR DESCRIPTION
**Pull request by hwchase17**

### CODE REVIEW

File: langchain/prompts/few_shot.py [hunk 1]
```diff
@@ -110,8 +110,14 @@ def format(self, **kwargs: Any) -> str:
         # Create the overall template.
         pieces = [self.prefix, *example_strings, self.suffix]
         template = self.example_separator.join([piece for piece in pieces if piece])
+        # Get partial params:
+        partial_kwargs = {
+            k: v if isinstance(v, str) else v()
+            for k, v in self.partial_variables.items()
+        }
+        all_kwargs = {**partial_kwargs, **kwargs}
```
Comment (agola11) [inline langchain/prompts/few_shot.py:118]: nit: can probs factor this out, it's in all 3 format functions

**Review by agola11 — APPROVED** [review]
lgtm, although how do we want to handle serializing prompt templates with partial_variables in play? I can see it being confusing to have one version of the prompt with partials and another one without

**Comment by hwchase17** [comment]
> lgtm, although how do we want to handle serializing prompt templates with partial_variables in play? I can see it being confusing to have one version of the prompt with partials and another one without

good point, ill add some check to stop serializing of partialed prompts

**Review by agola11 — APPROVED** [review]
looks great

**Pull request merged**